src/app/components/core/configuration-manager/configuration-manager.component.ts

```diff
@@ -229,6 +241,74 @@ export class ConfigurationManagerComponent implements OnInit {
         });
   }
 
+  /**
+   * Get edited south service configuration from show configuration page
+   * @param changedConfiguration changed configuration of a selected plugin
+   */
+  getChangedConfig(changedConfiguration: any, category: any) {
+    const cat = this.categoryDataCopy.find(cat => cat.key === category.key);
+    this.changedConfig = this.configurationControlService.getChangedConfiguration(changedConfiguration, cat);
+  }
+
+  save(catName: string) {
+    if (!isEmpty(this.changedConfig)) {
+      this.updateConfiguration(catName, this.changedConfig);
+    }
+  }
+
+  /**
+  * Update configuration
+  * @param categoryName Name of the cateogry
+  * @param configuration category updated configuration
+  */
+  updateConfiguration(categoryName: string, configuration: any) {
+    const files = this.getScriptFilesToUpload(configuration);
+    if (files.length > 0) {
+      this.uploadScript(categoryName, files);
+    }
+
+    if (!categoryName || isEmpty(configuration)) {
+      return;
+    }
```

Comment:
should not we move this up, before uploadScript block?

Reply:
No, in case of only file change of a config item, configuration value stays unchanged so no update config call but the file upload call only.